src/bdecode.cpp

```diff
@@ -479,7 +479,7 @@ namespace libtorrent
 		{
 			bdecode_token const& t = tokens[token];
 			TORRENT_ASSERT(t.type == bdecode_token::string);
-			int const size = m_root_tokens[token + 1].offset - t.offset - t.start_offset();
```

Comment:
I'd be happy with the patch to use ``tokens`` instead of ``m_root_tokens``. I think the patch as it stands now adds complexity and makes it less clear what the code is trying to achieve. If you think it's important that gcc 6.3 also generates the more efficient code, I would hope there would be a solution that's simple and clear as well